docs/current/cookbook/snippets/modules-shared-client/pipelines.mts

```diff
@@ -0,0 +1,11 @@
+import { Client, Container } from "@dagger.io/dagger"
+
+// get base image
+function base(client: Client): Container {
+  return client.container().from("alpine:latest")
+}
+
+// run command in base image
+export async function version(client: Client): Promise<string> {
+  return base(client).withExec(["cat", "/etc/alpine-release"]).stdout()
+}
```

Comment:
I think it can make even more sens to test it with a class, what about?
(I coded from scratch in the GH UI so the code may have mistake)

This could also be pretty similar for Go with a struct and Python with another class

```ts
import { Client, Container } from "@dagger.io/dagger"

class Pipeline {
    client: Client;

    constructor(client: Client) {
        this.client = client
    }

    private base(): Container {
       return this.client.container.().from("alpine:latest");
    }

    public async version(): Promise<string> {
        return this.base().withExec(["cat", "/etc/alpine-release"]).stdout();
    }
}
```